### PR TITLE
chore: Update create-release GitHub workflow to use 2 last minor Atlantis releases

### PR DESCRIPTION
**Pull request by vdmgolub**
Tested on `infracost/atlantis-test` images: https://hub.docker.com/u/infracost

### CODE REVIEW

**Comment by alikhajeh1** [comment]
@vdmgolub I think this PR should go into https://github.com/infracost/infracost-atlantis/pull/26, not master

File: .github/workflows/create-release.yml [hunk 1]
```diff
@@ -1,27 +1,61 @@
-name: Create release assets
+name: Create Atlantis release
 
-on:
-  push:
-    branches: [ master ]
+on: workflow_dispatch
```
Comment (vdmgolub) [inline .github/workflows/create-release.yml:3]: 1. Trigger on new CLI release via GH release workflow.
2. Trigger manually (automate later) when new Atlantis release is available.

File: .github/workflows/create-release.yml [hunk 2]
```diff
@@ -1,27 +1,61 @@
...
           username: ${{ secrets.DOCKER_USERNAME }}
           password: ${{ secrets.DOCKER_PASSWORD }}
-          repository: ${{ secrets.DOCKER_ORG }}/${{ secrets.DOCKER_REPOSITORY }}
-          tags: ${{ env.ATLANTIS_TAG }},latest
+
+      - name: Docker meta
+        id: meta
+        uses: docker/metadata-action@v3
+        with:
+          images: infracost/infracost-atlantis
```
Comment (vdmgolub) [inline .github/workflows/create-release.yml:44]: Hardcoded the name `DOCKER_ORG`/`DOCKER_REPOSITORY` are not secrets. Should we put it in an env var?

File: .github/workflows/create-release.yml [hunk 3]
```diff
@@ -1,27 +1,61 @@
...
+          flavor: |
+            latest=${{ matrix.latest }}
+          tags: |
+            type=raw,value=atlantis${{ matrix.atlantis_release }}-infracost${{ matrix.cli_release }}
+          labels: |
+            atlantis-version=${{ matrix.atlantis_version }}
+            cli-version=${{ matrix.cli_version }}
```
Comment (vdmgolub) [inline .github/workflows/create-release.yml:51]: Automate the Atlantis release checks:
1. Set a workflow to run on cron daily/weekly
2. Pull Atlantis versions like on line 17.
3. Pull images metadata and compare labels with Atlantis versions. This is a TBD step.
4. If they are newer, trigger the release workflow.

File: .github/workflows/create-release.yml (hunk 2 above)
Comment (tim775) [reply .github/workflows/create-release.yml:44]: hardcoding seems ok to me

File: .github/workflows/create-release.yml (hunk 3 above)
Comment (tim775) [reply .github/workflows/create-release.yml:51]: makes sense.

File: .github/workflows/create-release.yml [hunk 4]
```diff
@@ -1,27 +1,61 @@
...
+
+      - name: Docker meta
+        id: meta
+        uses: docker/metadata-action@v3
+        with:
+          images: infracost/infracost-atlantis
+          flavor: |
+            latest=${{ matrix.latest }}
+          tags: |
+            type=raw,value=atlantis${{ matrix.atlantis_release }}-infracost${{ matrix.cli_release }}
```
Comment (tim775) [inline .github/workflows/create-release.yml:48]: where is the `latest` tag generated (for `infracost-atlantis:latest`)?

**Review by tim775 — APPROVED** [review]

File: .github/workflows/create-release.yml [hunk 5]
```diff
@@ -1,27 +1,61 @@
...
+    runs-on: ubuntu-latest
+    outputs:
+      matrix: ${{ steps.set-matrix.outputs.matrix }}
+    steps:
+      - id: set-matrix
+        name: Set matrix
+        shell: bash
+        run: |
+          cli=$(curl -s https://api.github.com/repos/infracost/infracost/releases | jq -r '.[0].tag_name | capture("^v(?<major>0|[1-9]\\d*)\\.(?<minor>0|[1-9]\\d*)\\.(?<patch>0|[1-9]\\d*)") | {cli_version: "\(.major).\(.minor).\(.patch)", cli_release: "\(.major).\(.minor)"} | tostring')
+          atlantis=$(curl -s https://api.github.com/repos/runatlantis/atlantis/releases\?per_page\=100 | jq -r '[.[].tag_name] | map(capture("^v(?<major>0|[1-9]\\d*)\\.(?<minor>0|[1-9]\\d*)\\.(?<patch>0|[1-9]\\d*)")) | map({version: "\(.major).\(.minor).\(.patch)", group: "\(.major).\(.minor)", major: .major | tonumber, minor: .minor | tonumber, patch:  .patch | tonumber}) | group_by(.group) | map (.[0]) | sort_by(.major, .minor, .patch) | reverse | .[:2] | . as $result | map({atlantis_version: .version, atlantis_release: .group, latest: (.version == $result[0].version)}) | tostring')
```
[the remark this answers is not in the document]
Comment (vdmgolub) [reply .github/workflows/create-release.yml:17]: Oh yeah, Alistair showed me your thresholds example yesterday. Could save some time 🤦. But yes, jq is pure magic.

File: .github/workflows/create-release.yml (hunk 4 above)
Comment (vdmgolub) [reply .github/workflows/create-release.yml:48]: `matrix.latest` in `flavor` is true for Atlantis 0.18.1 and false for 0.17.6. `steps.meta.outputs.tags` will include `latest` automatically.

**Pull request merged**